src/ejabberd_service.erl

```diff
@@ -520,7 +520,7 @@ unregister_routes(StateData) ->
     ejabberd_router:unregister_components(Routes).
 
 get_routes(#state{host=Subdomain, is_subdomain=true}) ->
-    Hosts = ejabberd_config:get_global_option(hosts),
+    Hosts = ejabberd_config:get_global_option_or_default(hosts, []),
```

Comment:
This would only work for static hosts, I think it's better to fail loudly for now. We need to add support for dynamic domains to external components, it's needed, but let's do it separately.